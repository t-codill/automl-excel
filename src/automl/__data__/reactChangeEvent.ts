// tslint:disable: no-any
export const reactChangeEvent: React.ChangeEvent<any> = {
    target: {}
} as unknown as React.ChangeEvent<any>;
