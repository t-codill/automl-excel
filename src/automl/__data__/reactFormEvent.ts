// tslint:disable: no-any
export const reactFormEvent: React.FormEvent<any> = {
    preventDefault(): void { return; }
} as unknown as React.FormEvent<any>;
