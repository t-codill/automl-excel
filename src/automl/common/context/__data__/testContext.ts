import { PageNames } from "../../PageNames";
import { Logger } from "../../utils/logger";
import { IWorkspaceContextProps, WorkspaceError } from "../IWorkspaceProps";
import { WorkspaceFlight } from "../WorkspaceFlight";

export function getLogCustomEventSpy(): jest.SpyInstance<ReturnType<Logger["logCustomEvent"]>> {
    const spy = jest.spyOn(Logger.prototype, "logCustomEvent");
    spy.mockImplementation(jest.fn());
    return spy;
}
let pageName: PageNames = PageNames.Unknown;
export const testContext: IWorkspaceContextProps = {
    get pageName(): PageNames { return pageName; },
    get theme(): string { return "light"; },
    discoverUrls: {},
    flight: new WorkspaceFlight(""),
    location: "eastus",
    resourceGroupName: "testResource",
    subscriptionId: "00000000-0000-0000-0000-000000000000",
    logger: new Logger("test"),
    getToken: () => "testToken",
    setPageName: (value: PageNames) => { pageName = value; },
    workspaceName: "testWorkSpace",
    setNavigationBarButtons(): void { return; },
    onError: (err: WorkspaceError) => { throw err; },
    clearErrors: () => { return; },
    setHeader: () => { return; },
    setLoading: () => { return; }
};
