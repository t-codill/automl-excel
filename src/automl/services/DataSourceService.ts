import { ContainerURL, Models } from "@azure/storage-blob";
import { IServiceBaseProps } from "./ServiceBase";
import { IServiceBaseBlobProps, ServiceBaseBlob } from "./ServiceBaseBlob";

export class DataSourceService extends ServiceBaseBlob<ContainerURL> {
    constructor(props: IServiceBaseProps, blobProps: IServiceBaseBlobProps) {
        super(props, blobProps, ContainerURL);
    }
    public async listBlob(): Promise<Models.BlobItem[] | undefined> {

        return this.getAllWithNext<Models.BlobItem, Models.ListBlobsFlatSegmentResponse, string>
            (async (client, abortSignal) => {
                return client.listBlobFlatSegment(this.getAborter(abortSignal));
            },
                async (client, abortSignal, nextLink) => {
                    return client.listBlobFlatSegment(this.getAborter(abortSignal), nextLink);
                },
                (res) => res.segment.blobItems,
                (res) => res.nextMarker);
    }
}
