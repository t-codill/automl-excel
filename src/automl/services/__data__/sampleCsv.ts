export const sampleCsv = `a,b,c,d,e
1,2,3,4,5
1,2,3,4,5
1,2,3,4,5
1,2,3,4,5
1,2,3,4,5
`;

const largeCsvHeader = Array(50)
    .fill("aaaaa,bbbbb,ccccc,ddddd,eeeee")
    .join(",");

const largeCsvRow = Array(50)
    .fill("11111,22222,33333,44444,55555")
    .join(",");

export const largeCsv = `${largeCsvHeader}
${Array(50)
        .fill(largeCsvRow)
        .join("\n")}
`;
