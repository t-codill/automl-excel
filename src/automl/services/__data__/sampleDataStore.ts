import { AzureMachineLearningDatastoreManagementModels } from "@vienna/datastore";

export const sampleDataStore: Required<AzureMachineLearningDatastoreManagementModels.DataStoreDto> = {
    dataStoreType: "AzureBlob",
    name: "defaultDataStore",
    azureStorageSection: {
        accountKey: "key",
        accountName: "account",
        credentialType: "AccountKey",
        credential: "secret",
        containerName: "defaultContainer",
        isSas: false
    },
    azureDataLakeSection: {},
    azurePostgreSqlSection: {},
    azureSqlDatabaseSection: {},
    hasBeenValidated: true
};
