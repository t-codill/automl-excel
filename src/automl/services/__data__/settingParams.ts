import { AdvancedSetting } from "../AdvancedSetting";

export const settingParams: AdvancedSetting = {
    jobType: "classification",
    column: "1",
    metric: "AUC_weighted",
    experimentTimeoutMinutes: 60,
    maxIteration: 15,
    experimentExitScore: 0.99,
    maxConcurrent: 1,
    maxCores: 2,
    preprocess: true,
    nCrossValidations: 5,
    blacklistAlgos: null,
    validationSize: 0.2,
    grainColumns: null,
    maxHorizon: null,
    timeSeriesColumn: null
};
