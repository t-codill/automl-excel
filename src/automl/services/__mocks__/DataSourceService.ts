import { Models } from "@azure/storage-blob";
import { sampleStorageBlob } from "../../../__data__/sampleStorageBlob";

export class DataSourceService {
    public async listBlob(): Promise<Models.BlobItem[] | undefined> {

        return [sampleStorageBlob, sampleStorageBlob];
    }

    public dispose(): void {
        return;
    }

    public reset(): void {
        return;
    }
}
