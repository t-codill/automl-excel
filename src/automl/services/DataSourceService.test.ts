import { sampleStorageAccount } from "../../__data__/sampleStorageAccount";
import { sampleStorageContainer } from "../../__data__/sampleStorageContainer";
import { testContext } from "../common/context/__data__/testContext";
import { DataSourceService } from "./DataSourceService";

jest.mock("@azure/storage-blob");

const service = new DataSourceService(testContext, {
    account: sampleStorageAccount,
    container: sampleStorageContainer,
    sasToken: "sampleSasToken"
});

describe("DataSourceService", () => {
    it("should listBlob", async () => {
        const result = await service.listBlob();
        expect(result)
            .toMatchSnapshot();
    });
});
