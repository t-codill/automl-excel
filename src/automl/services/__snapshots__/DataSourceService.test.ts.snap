// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DataSourceService should listBlob 1`] = `
Array [
  Object {
    "deleted": false,
    "name": "sampleBlob",
    "properties": Object {
      "contentLength": 1000,
      "etag": "sampleEtag",
      "lastModified": 2019-01-01T00:00:00.000Z,
    },
    "snapshot": "sampleSnapShot",
  },
  Object {
    "deleted": false,
    "name": "sampleBlob",
    "properties": Object {
      "contentLength": 1000,
      "etag": "sampleEtag",
      "lastModified": 2019-01-01T00:00:00.000Z,
    },
    "snapshot": "sampleSnapShot",
  },
]
`;
